tests/cupy_tests/random_tests/test_sample.py

```diff
@@ -47,6 +55,26 @@ def test_bound_2(self):
         self.assertEqual(min(vals), 0)
         self.assertEqual(max(vals), 1)
 
+    @condition.repeat(3, 10)
+    def test_bound_overflow(self):
+        # 100 - (-100) exceeds the range of int8
+        val = random.randint(numpy.int8(-100), numpy.int8(100), size=20).get()
+        self.assertEqual(val.shape, (20,))
+        self.assertGreaterEqual(val.min(), -100)
+        self.assertLess(val.max(), 100)
+
+    @condition.repeat(3, 10)
+    def test_bound_float1(self):
+        # generate floats s.t. int(low) < int(high)
+        low, high = sorted(numpy.random.uniform(-5, 5, size=2))
+        low -= 1
+        high += 1
```

Comment:
This logic only generates pairs at least 2.0 apart and thus not exhaustive enough.
Actually, the specification of allowed spans in NumPy looks a bit complicated around zero...
* -2 < low < -1 ⇒ -1 < high
* low == -1 ⇒ -1 < high
* -1 < low < 0 ⇒ 1 <= high 
* low == 0 ⇒ 1 <= high
* 0 < low < 1 ⇒ 1 <= high
* low == 1 ⇒ 1 < high
* 1 < low < 2 ⇒ 2 <= high

I think it's better to test on these 7 configurations explicitly, instead of (or additionally to) using random values.